Formula/c/clang-uml.rb

```diff
@@ -0,0 +1,123 @@
+class ClangUml < Formula
+  desc "Customizable automatic UML diagram generator for C++ based on Clang"
+  homepage "https://github.com/bkryza/clang-uml"
+  url "https://github.com/bkryza/clang-uml/archive/refs/tags/0.5.3.tar.gz"
+  sha256 "e830363ec510f14cc738c6509107b3f52bc55ececc2e27c068cadb093604e943"
+  license "Apache-2.0"
+  head "https://github.com/bkryza/clang-uml.git", branch: "master"
+
+  depends_on "cmake" => [:build, :test]
+  depends_on "pkg-config" => :build
+  depends_on "llvm"
+  depends_on "yaml-cpp"
+
+  fails_with gcc: "5"
+
+  def llvm
+    deps.map(&:to_formula)
+        .find { |f| f.name.match?(/^llvm(@\d+)?$/) }
+  end
+
+  def install
+    args = %W[
+      -DBUILD_TESTS=OFF
+      -DCLANG_UML_ENABLE_BACKTRACE=OFF
+      -DCMAKE_INSTALL_RPATH=#{rpath(source: libexec/"bin", target: llvm.opt_lib)}
+    ]
+
+    # If '.git' directory is not available during build, we need
+    # to provide the version using a CMake option
+    args.append("-DGIT_VERSION=#{version}") unless build.head?
+
+    # clang-uml needs to be linked to LLVM c++ when compiled with Clang, otherwise
+    # Trace/BPT trap is raised whenever an exception is thrown (even if it's handled properly)
+    if OS.mac?
+      cc_bin = llvm.opt_bin/"clang"
+      cxx_bin = llvm.opt_bin/"clang++"
+      args.append(%Q(-DCMAKE_EXE_LINKER_FLAGS="-L#{llvm.opt_lib}/c++ -Wl,-rpath,#{llvm.opt_lib}/c++"))
+    end
+
+    with_env(CC: cc_bin, CXX: cxx_bin) do
+      system "cmake", "-S", ".", "-B", "build", *std_cmake_args(install_prefix: libexec), *args
+    end
```

Comment:
Can't we use the system compiler while retaining `CMAKE_EXE_LINKER_FLAGS`?

Reply:
@carlocab Thanks for the comment! I've tried that, but I get the same errors. In a nutshell the problem I'm having here is that when `clang-uml` is built on macos{13,14}/arm64, there is some issue with linkage or ABI compatibility between libc++ and libLLVM (I think), which results in the following error that is raised whenever an exception is thrown within the `clang-uml` process (even if it's for a legitimate reason): 
```console
  ==> /opt/homebrew/Cellar/clang-uml/0.5.3/bin/clang-uml --help 2>&1
  sh: line 1: 13258 Trace/BPT trap: 5       /opt/homebrew/Cellar/clang-uml/0.5.3/bin/clang-uml --help 2>&1
  
  Error: clang-uml: failed
  ::error::clang-uml: failed
  An exception occurred within a child process:
    Minitest::Assertion: Expected: 0
    Actual: 133
```

Now, I know it is possible to make it work as it works in my Github Actions worker (macos 14/arm64) here: https://github.com/bkryza/clang-uml/actions/runs/9919731556/job/27405865335

I will have access to a macos14/M2 tomorrow so hopefully I'll be able to resolve it...

Reply:
I wonder if there's some sort of mismatch between the header and library versions. This is possible, for example, if `clang` uses the system `libc++` headers but links to the LLVM formula's `libc++` (or vice-versa).

I'd have thought compiling using LLVM's `clang` would have resolved this, though (since LLVM's `clang` typically prioritises the bundled `libc++` headers instead of using the system's).

Reply:
@carlocab I think you're right, the thing is I didn't manage to link against LLVM's provided libstdc++, the final binary always uses the system one, even though I added the `-DCMAKE_EXE_LINKER_FLAGS="-L/opt/homebrew/opt/llvm/lib/c++ -Wl,-rpath,/opt/homebrew/opt/llvm/lib/c++"` linker flags as suggested by `llvm`'s caveat message.

It can be seen in the `brew linkage` logs:
```
 ==> brew linkage --cached clang-uml
System libraries:
  /usr/lib/libSystem.B.dylib
  /usr/lib/libc++.1.dylib
Homebrew libraries:
  /opt/homebrew/opt/llvm/lib/libLLVM.dylib (llvm)
  /opt/homebrew/opt/llvm/lib/libclang-cpp.dylib (llvm)
  /opt/homebrew/opt/llvm/lib/libunwind.1.dylib (llvm)
  /opt/homebrew/opt/yaml-cpp/lib/libyaml-cpp.0.8.dylib (yaml-cpp)
```

I'll try to resolve this tomorrow using `brew install --interactive` on arm64 macos 14...

Reply:
Found a fix for you: https://github.com/Homebrew/homebrew-core/pull/177569#discussion_r1681458435

If you apply all my suggestions I think this should be ready to go.

Reply:
@carlocab Wow, thank you! I've just pushed the fixes to all your comments...